test/integration/tests/policyauthorize.sh

```diff
@@ -0,0 +1,135 @@
+file_verifying_key_name="verifying_key_name"
+file_verifying_key_ctx="verifying_key_ctx"
+file_policyref="policyref"
+
+onerror() {
+    echo "$BASH_COMMAND on line ${BASH_LINENO[0]} failed: $?"
+    exit 1
+}
+trap onerror ERR
+
+cleanup() {
+  rm -f  $file_pcr_value $file_policy $file_session_file $file_private_key \
+    $file_public_key $file_dummy_verification_ticket $file_verifying_key_public \
+    $file_verifying_key_name $file_verifying_key_ctx $file_policyref \
+    $file_authorized_policy_1 $file_authorized_policy_2
+
+  tpm2_flushcontext -S $file_session_file 2>/dev/null || true
+}
+trap cleanup EXIT
+
+start_up
+
+cleanup
+
+generate_policyAuthorize () {
+
+  tpm2_startauthsession -Q -S $file_session_file
+
+  tpm2_policyauthorize -Q \
+  -S $file_session_file -d $file_dummy_verification_ticket \
+  -f $3 -a $1 -b $2 -c $4
+   
+  tpm2_flushcontext -S $file_session_file 
+}
+
+openssl genrsa -out $file_private_key 2048 2>/dev/null
+openssl rsa -in $file_private_key -out $file_public_key -pubout 2>/dev/null
+
+#HACK: Ideally enable loadexternal tool to directly load a public pem 
+openssl asn1parse -noout -in $file_public_key -strparse 19 -out modulus.bin 
+echo "01160001000b000600720000001000100800000000000100" | xxd -r -p > tss_header.bin
+dd if=modulus.bin of=temp1 bs=1 count=256 skip=9 2>/dev/null
+cat tss_header.bin temp1 > $file_verifying_key_public
+rm tss_header.bin modulus.bin temp1
+tpm2_loadexternal -a o -u $file_verifying_key_public \
+-o $file_verifying_key_ctx 2>/dev/null
+
+tpm2_readpublic -c $file_verifying_key_ctx | grep 'name:' |\
+grep -v q | awk '{print $2}' | xxd -r -p > $file_verifying_key_name
+
+dd if=/dev/urandom of=$file_policyref bs=1 count=32 2>/dev/null
+
+tpm2_pcrlist -Q -L ${alg_pcr_policy}:${pcr_ids} -o $file_pcr_value
+tpm2_startauthsession -Q -S $file_session_file
+tpm2_policypcr -Q -S $file_session_file -L ${alg_pcr_policy}:${pcr_ids} -F $file_pcr_value -f $file_policy
+tpm2_flushcontext -S $file_session_file
+
+echo "8022400000010030b1923e84f25626f3d7c25a4e9d3ef6b88cb3e1807e3e\
+  b7f0102bacc68252f548093c511fbe7687ed45c0e2c10a7fe469"\
+  | xxd -r -p > $file_dummy_verification_ticket
+generate_policyAuthorize $file_policy $file_policyref $file_authorized_policy_1 \
+  $file_verifying_key_name
+
+tpm2_pcrextend  \
+  0:sha256=e7011b851ee967e2d24e035ae41b0ada2decb182e4f7ad8411f2bf564c56fd6f
+
+tpm2_pcrlist -Q -L ${alg_pcr_policy}:${pcr_ids} -o $file_pcr_value
+tpm2_startauthsession -Q -S $file_session_file
+tpm2_policypcr -Q -S $file_session_file -L ${alg_pcr_policy}:${pcr_ids} -F $file_pcr_value -f $file_policy
+tpm2_flushcontext -S $file_session_file
+
+# negative test, clear the error handler
+trap - ERR
+
+generate_policyAuthorize $file_policy $file_policyref $file_authorized_policy_2 \
+  $file_verifying_key_name
+
+# restore the error handler
+trap onerror ERR
+
+diff $file_authorized_policy_1 $file_authorized_policy_2
+
+rc=$?
+
+if [ $rc -eq 1 ]; then
```

Comment:
this would never run, as line 127 fails tne ERR trap is invoked, so you can just remove this. You should probably move your negative test on line 121 to the end, so you don't need to clear and reset the trap handler.